migrations/20220607041349_add_mfa_schema.up.sql

```diff
@@ -0,0 +1,51 @@
+-- See: https://stackoverflow.com/questions/7624919/check-if-a-user-defined-type-already-exists-in-postgresql/48382296#48382296
+DO $$ BEGIN
+    CREATE TYPE factor_type AS ENUM('totp', 'webauthn');
+    CREATE TYPE factor_status AS ENUM('disabled', 'unverified', 'verified');
+EXCEPTION
+    WHEN duplicate_object THEN null;
+END $$;
+
+-- auth.mfa_factors definition
+CREATE TABLE IF NOT EXISTS auth.mfa_factors(
+       id VARCHAR(256) NOT NULL,
+       user_id uuid NOT NULL,
+       friendly_name VARCHAR(256) NULL,
+       factor_type factor_type NOT NULL,
+       status factor_status NOT NULL,
+       created_at timestamptz NOT NULL,
+       updated_at timestamptz NOT NULL,
+       secret_key VARCHAR(256) NOT NULL,
+       UNIQUE(user_id, friendly_name),
+       CONSTRAINT mfa_factors_pkey PRIMARY KEY(id),
+       CONSTRAINT mfa_factors_user_id_fkey FOREIGN KEY (user_id) REFERENCES auth.users(id) ON DELETE CASCADE
+);
+comment on table auth.mfa_factors is 'Auth: stores metadata about factors';
+
+-- auth.mfa_challenges definition
+CREATE TABLE IF NOT EXISTS auth.mfa_challenges(
+       id VARCHAR(256) NOT NULL,
+       factor_id VARCHAR(256) NOT NULL,
+       created_at timestamptz NOT NULL,
+       verified BOOLEAN NOT NULL,
```

Comment:
if you think it's necessary to know when the challenge was verified then a timestamp field might be better here - ( e.g. `verified_at timestamptz`, a null value would indicate that the challenge is not verified yet)